contracts/exchange/Orderbook.sol

```diff
@@ -139,6 +144,40 @@ contract Orderbook is IOrderbook, ManagerBase {
                 orderAmounts[i] = orders[_orderIds[i]].amountOrdered - orders[_orderIds[i]].amountFilled;
             }
         }
+
+        // get amounts ordered based on AmountToFill and Max Spend
+        amountFilled = 0;
+        uint256 amountSpentOnOrder = 0;
+        for (uint256 i = 0; i < orderAmounts.length; ++i) {
+            if (orderAmounts[i] > 0) {
+                amountSpentOnOrder = orders[_orderIds[i]].price * orderAmounts[i];
+                
+                // Check if there's you're still under tha max Spend
+                if (maxSpend == 0) {
+                    orderAmounts[i] = 0;
+                    continue;
+                } else if (maxSpend > amountSpentOnOrder) {
+                    maxSpend -= amountSpentOnOrder;
+                } else if (maxSpend < amountSpentOnOrder) {
+                    orderAmounts[i] = maxSpend / orders[_orderIds[i]].price;
```

Comment:
Is it possible for price to be zero? Maybe we need a require price >= 0 clause in here.

Reply:
i don't know why, but it seems my comments are gone somehow. Reposting comment: 

Price cannot be set to zero. In PlaceOrder(), there's a check to make sure amounts and price is never set to zero. So if the order exists, the price cannot be zero.